### ISSUE TITLE
akeneo-fpm image fails at startup

### ISSUE THREAD

**Issue by Erwyn** [akeneo/Dockerfiles#123]
The akeneo-fpm gives this back to me when issuing a `docker-compose up akeneo`:

```
Starting pimcommunitydev_mongodb_1
Starting pimcommunitydev_mysql_1
Starting pimcommunitydev_akeneo_1
Attaching to pimcommunitydev_akeneo_1
akeneo_1         | [19-Jan-2017 14:18:18] ERROR: Unable to create the PID file (/run/php/php7.1-fpm.pid).: No such file or directory (2)
akeneo_1         | [19-Jan-2017 14:18:18] ERROR: FPM initialization failed
pimcommunitydev_akeneo_1 exited with code 78
```

**Comment by damien-carcel** [comment]
Hello @Erwyn.

This is a problem of initialization order, in the `depends_on` section of the compose file. I have inverted the dependencies of the `nginx` images: the `nginx-behat` one depends on `akeneo`, and `nginx` one on `akeneo-behat` when it should be the other way.

I will investingate it further and fix the compose file example I provide.

**Comment by Erwyn** [comment]
Oh, but then this may be my bad as well, because I ran it using the `docker-compose up akeneo` command thus ignoring the nginx container.

**Comment by damien-carcel** [comment]
Oh, indeed. I always use `docker-composer up -d`, so I can't tell the result of your command ;-). But anyway, it doesn't change the fact that my compose example is wrong!

**Comment by Erwyn** [comment]
Okay so after using the normally right command `docker-compose up nginx`, this is what I get:

```
❯ docker-compose up nginx
Creating network "pimcommunitydev_akeneo" with the default driver
Creating network "pimcommunitydev_behat" with the default driver
Creating pimcommunitydev_mongodb_1
Creating pimcommunitydev_mysql_1
Creating pimcommunitydev_akeneo_1
Creating pimcommunitydev_nginx_1
Attaching to pimcommunitydev_nginx_1
nginx_1         | 2017/01/19 15:34:23 [emerg] 9#9: host not found in upstream "akeneo" in /etc/nginx/conf.d/default.conf:11
nginx_1         | nginx: [emerg] host not found in upstream "akeneo" in /etc/nginx/conf.d/default.conf:11
pimcommunitydev_nginx_1 exited with code 1
```

**Comment by damien-carcel** [comment]
I can reproduce with `docker compose up -d` too, as nginx and fpm images are both with status "EXIT".
I think they don't wait for each other to be ready to communicate.

**Comment by damien-carcel** [comment]
The problem has been fixed by #125. The issue was only for PHP 7.1 (branches ’php-7.1’ and ’master’). For those versions, the FPM process needs a folder ’/run/php’ to create its socket or its process identifier (according to how FPM is configured).

The problem is solved by creating this folder in the ’carcel/fpm’ dockerfile before launching the process (this impacts ’carcel/akeneo-fpm’ and ’carcel/akeneo-behat-fpm’ images too). php-7.1 images are already built, and master images (latest) are finishing as I write.

It does not affect PHP 5.6 and 7.0 images, as they are natively configured to create sockets and PID directly in ’/run’.

I also fixed the compose file example.

Let me know if this works for you.

Regards,
Damien
